applications/procedural_narrator/tests/procedural_narrator_functional.py

```
@@ -0,0 +1,230 @@
+            prefix, encoded = data_url.split(",", 1)
+            if prefix != "data:video/mp4;base64":
+                raise ValueError("request did not contain an MP4 data URL")
+            video = base64.b64decode(encoded, validate=True)
+            if len(video) < 12 or video[4:8] != b"ftyp":
+                raise ValueError("request did not contain a valid MP4 header")
+            if payload.get("stream") is not True:
+                raise ValueError("request did not enable streaming")
+            if payload.get("chat_template_kwargs") != {"enable_thinking": False}:
+                raise ValueError("request did not disable thinking mode")
+            self.server.request_payload = payload
+        except Exception as error:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)